addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.fields import Command
+from odoo.osv import expression
+from odoo.tools import html2plaintext, single_email_re, email_normalize
+
+from collections import defaultdict
+
+
+class ChatbotScriptStep(models.Model):
+    _name = 'im_livechat.chatbot.script_step'
+    _description = 'Chatbot Script Step'
+    _order = 'sequence, id'
+
+    message = fields.Text(string='Message', required=True)
+    sequence = fields.Integer(string='Sequence')
+    chatbot_id = fields.Many2one(
+        'im_livechat.chatbot.script', string='Chatbot', required=True, ondelete='cascade')
+    step_type = fields.Selection([
+        ('text', 'Text'),
+        ('question_selection', 'Question'),
+        ('question_email', 'Email'),
+        ('question_phone', 'Phone'),
+        ('forward_operator', 'Forward to Operator'),
+    ], default='text', required=True)
+    # answers
+    answer_ids = fields.One2many(
+        'im_livechat.chatbot.script_question_answer', 'step_id',
+        copy=True, string='Answers')
+    triggering_answer_ids = fields.Many2many(
+        'im_livechat.chatbot.script_question_answer', 'chatbot_script_chatbot_script_question_answer_rel',
+        domain="[('step_id.sequence', '<', sequence)]",
+        copy=False,  # copied manually, see im_livechat.chatbot.script#copy
+        string='Only If', help='Show this step only if all of these answers have been selected.')
+    # forward-operator specifics
+    is_forward_operator_child = fields.Boolean(compute='_compute_is_forward_operator_child')
+
+    @api.onchange('sequence')
+    def onchange_sequence(self):
```

Comment:
Ok I got the idea ... sequence is an important field then.

If we want that, I would do a computed field, and add a constraint in python. Relying on an onchange is not really a good idea, we may end up with strange configuration.